src/IO/H5/ContiguousVolumeData.cpp

```diff
@@ -0,0 +1,250 @@
+    component_names.push_back(component_name);
+  }
+  // Extract Tensor Data one component at a time
+  std::vector<size_t> total_extents;
+  std::string grid_names;
+  std::vector<int> total_connectivity;
+  // Need to keep track the total number of points inserted into the
+  // connectivity after each iteration to be sure each point gets a
+  // unique representation in the topology data
+  int point_counter = 0;
+  bool First = true;  // True if first tensor component being accessed
+  for (size_t i = 0; i < component_names.size(); i++) {
+    std::string component_name = component_names[i];
+    std::vector<double> tensor_data{};
+    for (auto& element : elements) {
+      if (First) {
+        // Get the name of the element
+        const auto& first_tensor_name = element.tensor_components.front().name;
+        ASSERT(first_tensor_name.find_last_of('/') != std::string::npos,
+               "The expected format of the tensor component names is "
+               "'GROUP_NAME/COMPONENT_NAME' but could not find a '/' in '"
+                   << first_tensor_name << "'.");
+        const auto spatial_name =
+            first_tensor_name.substr(0, first_tensor_name.find_last_of('/'));
+        grid_names += spatial_name + ":";
+        // Process the element extents
+        auto extents = element.extents;
+        total_extents.insert(total_extents.end(), extents.begin(),
+                             extents.end());
+
+        // Find the number of points in the local connectivity
+        int local_numpoints = [&extents]() {
+          int numpoints = 1;
+          for (int length : extents) {
+            numpoints *= length;
+          }
+          return numpoints;
+        }();
+        // Generate the connectivity data for the element
+        const std::vector<int> connectivity =
+            [&extents, &point_counter ]() noexcept {
+          std::vector<int> local_connectivity;
+          for (const auto& cell : vis::detail::compute_cells(extents)) {
+            for (const auto& bounding_indices : cell.bounding_indices) {
+              local_connectivity.emplace_back(
+                  point_counter + static_cast<int>(bounding_indices));
+            }
+          }
+          return local_connectivity;
+        }
+        ();
+        point_counter += local_numpoints;
+        total_connectivity.insert(total_connectivity.end(),
+                                  connectivity.begin(), connectivity.end());
+
+      }  // if (First)
+      auto tensor_component = element.tensor_components[i];
+      DataVector local_tensor_data = tensor_component.data;
+      tensor_data.insert(tensor_data.end(), local_tensor_data.begin(),
+                         local_tensor_data.end());
+
+    }  // for each element
+
+    // Write the data for the tensor component
+    std::vector<size_t> Size{tensor_data.size()};
+    h5::write_data(observation_group.id(), tensor_data, Size, component_name);
```

Comment:
`Size` -> `std::vector<size_t>{tensor_data.size()}` (you might even be able to do `, tensor_data, {tensor_data.size()}, component_name`